### DISCUSSION TITLE
[Feature] add-photos-to-albums-by-path-and-name-using-api

### DISCUSSION THREAD

**Discussion by andyxpert** [immich-app/immich#13261] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

As the title states, could you add an api method that allows adding a list of photos to a specific album by their path and filename ?

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
That sounds awfully specific. What are you trying to do that the current api doesn't allow?

**Reply by andyxpert** [reply]
I created a few albums in the past, then downloaded them - now I have a folder with the actual photos... and I want to re-create the album with those exact photos.
Basically I want to create albums with specific photos (for which I have the files/fileNames) without having to find them in the web interface individually and add them one by one to the album.

I can use the search/metadata API method to find the assets, I get their ids, but when i try to add them to the album with API call albums/:albumId/assets with this as a body, it fails saying the id isn't correct:
'{"ids": [xxxxxx-xxxx-xxx-xxx-xxxxxxx]}'
so maybe it's a problem with that API method that adds assets to albums ? though i doubt it... am I doing something wrong ?
I just wanted a simplified version of "adding photos to albums by the filename" which I guess it's a fairly common use-case

**Reply by bo0tzz** [reply]
The immich cli has an `--album` flag that should do what you want, no need for any custom scripts here.

**Comment by andyxpert** [comment]
CLI uploads new versions, I already have my photos in a read only external library and I wanted those to be added to albums

Solved it by:
- listing the file names
- created the immich web search urls for them
- opened all the URLs in browser
- added the found image(s) to the desired albums

It was faster like this and allowed me to verify the image before adding it to the album.

Thread can be closed.
